### PR TITLE
Use the proper charset for decoding HTTP responses

### PR DESCRIPTION
**Pull request by abarth**
Previously we always used Latin-1.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @collinjackson

**Comment by collinjackson** [comment]
lgtm

File: packages/flutter/lib/src/http/response.dart [hunk 1]
```diff
@@ -2,23 +2,20 @@
...
 class Response {
   /// Creates a [Response] object with the given fields.
   ///
   /// If [bodyBytes] is non-null, it is used to populate [body].
-  Response({
-    Uint8List bodyBytes,
-    this.statusCode
-  }) : body = bodyBytes != null ? new String.fromCharCodes(bodyBytes) : null,
-       bodyBytes = bodyBytes;
+  Response.bytes(this.bodyBytes, this.statusCode, { this.headers: const {} });
```
Comment (Hixie) [inline packages/flutter/lib/src/http/response.dart]: types on the literal

File: packages/flutter/lib/src/http/response.dart [hunk 2]
```diff
@@ -2,23 +2,20 @@
...
 
   /// The result of decoding [bodyBytes] using ISO-8859-1.
   ///
   /// If [bodyBytes] is null, this will also be null.
-  final String body;
+  String get body => _encodingForHeaders(headers).decode(bodyBytes);
```
Comment (Hixie) [inline packages/flutter/lib/src/http/response.dart]: this should be a function, not a getter, since it is expensive

Comment (abarth) [inline packages/flutter/lib/src/http/response.dart]: I'm just matching the API from `package:http`.  I agree that it's a misleading API.

File: packages/flutter/lib/src/http/response.dart [hunk 3]
```diff
@@ -27,4 +24,67 @@ class Response {
...
+  return c == ' ' || c == '\t' || c == '\n' || c == '\r' || c == '\f';
+}
+
+int _skipSpaces(String string, int index) {
+  while (index < string.length && _isSpace(string[index]))
+    index += 1;
+  return index;
+}
+
+// https://html.spec.whatwg.org/#algorithm-for-extracting-a-character-encoding-from-a-meta-element
```
Comment (Hixie) [inline packages/flutter/lib/src/http/response.dart]: fwiw I'm not convinced that this is the right algorithm, but from my phone I am having no success in finding the right one, so...

**Comment by eseidel** [comment]
Turns out this broke iOS: https://github.com/flutter/flutter/issues/3486  Our drive tests caught it, but the bots had issues and we had to disable the drive tests for other reasons. :(

**Pull request merged**